src/app/component/shared/components/calendar-base/calendar-base.component.ts

```diff
@@ -28,18 +29,19 @@ export class CalendarBaseComponent implements OnInit, OnDestroy {
   public currentMonth = new Date().getMonth();
   public currentYear = new Date().getFullYear();
 
-  public isHabitsPopUpOpen;
-  public habitsCalendarSelectedDate;
   public selectedDay;
   public isDayTracked;
   public habits;
   public isFetching;
-  public calendarIcons;
   public checkIfFuture;
   public toggleEnrollHabit;
 
   public langChangeSub: Subscription;
   public defaultTranslateSub: Subscription;
+  private destroyGetHabits = new Subject<void>();
+  public destroyOpenPopup = new Subject<void>();
+  public destroyEnroll = new Subject<void>();
+  public destroyUnEnroll = new Subject<void>();
```

Comment:
hi, you need only 1 destroy Subject for all Observables.
https://medium.com/impact-developers/how-to-destroy-observables-in-angular-313dec343b45

Reply:
Thanks. Fixed.